doc/sf/logaddexp.qbk

```diff
@@ -0,0 +1,139 @@
+[/
+  Copyright Matt Borland 2022
+  Distributed under the Boost Software License, Version 1.0.
+  (See accompanying file LICENSE_1_0.txt or copy at
+  http://www.boost.org/LICENSE_1_0.txt).
+]
+
+[section:logaddexp logaddexp and logsumexp]
+
+[h4 Synopsis]
+
+    #include <boost/math/special_functions/logaddexp.hpp>
+
+    namespace boost { namespace math {
+
+    template <typename Real>
+    Real logaddexp (Real x1, Real x2);
+
+    }} // namespaces
+
+    #include <boost/math/special_functions/logsumexp.hpp>
+
+    namespace boost { namespace math {
+
+    template <typename ForwardIterator, typename Real = std::iterator_traits<ForwardIterator>::value_type>
+    Real logsumexp (ForwardIterator first, ForwardIterator last);
+
+    template <typename ForwardContainer, typename Real = ForwardContainer::value_type>
+    Real logsumexp (const ForwardContainer& c);
+
+    template <typename... Args, typename Real = typename std::common_type<Args...>::type>
+    Real logsumexp(Args&& ...args)
+
+    }} // namespace
+
+
+The function `logaddexp` computes log(exp(x1) + exp(x2)).
+The function `logsumexp` is generalized to compute log(exp(x1) + exp(x2) + ...).
+This function is useful in statistics where the calculated probabilities of events may be so small as to exceed the range of normal floating point numbers. 
+In such cases the logarithm of the calculated probability is stored.
```

Comment:
Might want to cite that Higham blog post because that's the formula we use to compute `logsumexp`.